### PR TITLE
improve batch crypto sanity checks error reporting

### PR DESCRIPTION
**Pull request by mratsim**
Motiviated by debugging REST aggregation failures.

### CODE REVIEW

File: beacon_chain/gossip_processing/batch_validation.nim [hunk 1]
```diff
@@ -331,7 +332,7 @@ proc scheduleAggregateChecks*(
               )
   if not sig.isSome():
     batchCrypto.scheduleBatch(fresh)
-    return none(R)
+    return err("Attestation batch validation: no attester found or invalid pubkey found.")
```
Comment (arnetheduck) [inline beacon_chain/gossip_processing/batch_validation.nim:335]: why not return the `Result` all the way from `addAttestation`? then we can have an even more specific error

**Review by arnetheduck — APPROVED** [review]

File: beacon_chain/gossip_processing/batch_validation.nim [hunk 2]
```diff
@@ -251,17 +251,18 @@ proc scheduleAttestationCheck*(
...
       fork: Fork, genesis_validators_root: Eth2Digest,
       epochRef: EpochRef,
       signedAggregateAndProof: SignedAggregateAndProof
-     ): Option[(
-       tuple[slotCheck, aggregatorCheck, aggregateCheck:
-         Future[BatchResult]],
-       CookedSig)] =
+     ): Result[
+          (
+            tuple[slotCheck, aggregatorCheck, aggregateCheck: Future[BatchResult]],
```
Comment (zah) [inline beacon_chain/gossip_processing/batch_validation.nim:263]: What is the benefit of nesting a tuple within another tuple here (as opposed to flattening to a single tuple)?

File: beacon_chain/gossip_processing/batch_validation.nim (hunk 1 above)
Comment (mratsim) [reply beacon_chain/gossip_processing/batch_validation.nim:335]: I thought about that but it required changing more APIs in many places.

File: beacon_chain/gossip_processing/batch_validation.nim (hunk 2 above)
Comment (mratsim) [reply beacon_chain/gossip_processing/batch_validation.nim:263]: I'm not too sure, I just reused the original CookedSig tuple as-is. I'm agnostic to whichever solution.

File: beacon_chain/gossip_processing/batch_validation.nim (hunk 1 above)
Comment (arnetheduck) [reply beacon_chain/gossip_processing/batch_validation.nim:335]: I suspect it's worth it, in the end - there aren't _that_ many addAttestation calls after all

**Comment by zah** [comment]
OK, let's merge this to facilitate debugging of the failing REST APIs. The code style comments can be addressed in a follow-up PR.

**Pull request merged**